server/src/infra/ldap_handler.rs

```diff
@@ -120,15 +121,51 @@ fn get_user_attribute(user: &User, attribute: &str, dn: &str) -> Result<Option<V
     }))
 }
 
+fn expand_attribute_wildcards(attributes: &[String], all_attribute_keys: &[&str]) -> Vec<String> {
+    let mut attributes_out = attributes.to_owned();
+
+    if attributes_out.iter().any(|x| x == "*") || attributes_out.is_empty() {
+        debug!(r#"Expanding *"#);
+        // Splice in all non-operational attributes
+        attributes_out.extend(all_attribute_keys.iter().map(|s| s.to_string()));
+        // Remove occurrences of '*'
+        attributes_out.retain(|x| x != "*");
+    }
+
+    if attributes_out.iter().any(|x| x == "+") {
+        debug!(r#"Removing +"#);
+        attributes_out.retain(|x| x != "+");
+    }
+
+    debug!(r#"Expanded: "{:?}""#, &attributes_out);
+
+    // Deduplicate, preserving order
+    attributes_out = attributes_out.into_iter().unique().collect_vec();
+
+    return attributes_out;
```

Comment:
Just `attributes_out`, no return, no semicolon. Or same on the line above: 
`attributes_out.into_iter().unique().collect_vec()`

Reply:
Oh - is this lint given by `cargo clippy` ?

Reply:
Indeed! And it's also nice Rust style. In general, keep return for exiting early from the function.